MANIFEST.in

```
@@ -15,4 +15,18 @@
 # specific language governing permissions and limitations
 # under the License.
 
-graft src
+# Include Cython source files for building from source
+recursive-include pyiceberg *.pyx *.c
+
+# Include test files in sdist
+recursive-include tests *.py
+
+# Include development files
+include Makefile
+recursive-include dev *
+
+# Exclude build artifacts
+prune .venv
+prune build
+prune dist
+prune .pytest_cache
```

```suggestion
prune .pytest_cache
prune pyiceberg.egg-info
prune setup.cfg
```
can we get rid of these `pyiceberg.egg-info` and `setup.cfg` from sdist? These are auto generated and we can also remove them from `dev/.rat-excludes`